modules/planning/tasks/optimizers/open_space_trajectory_partition/open_space_trajectory_partition.cc

```diff
@@ -17,21 +17,406 @@
+
+    // Check if have reached endpoint of trajectory
+    const TrajectoryPoint& trajectory_end_point = trajectory.back();
+
+    const PathPoint& path_end_point = trajectory_end_point.path_point();
+
+    double distance_to_trajs_end =
+        std::sqrt((path_end_point.x() - vehicle_state.x()) *
+                      (path_end_point.x() - vehicle_state.x()) +
+                  (path_end_point.y() - vehicle_state.y()) *
+                      (path_end_point.y() - vehicle_state.y()));
+
+    double traj_point_moving_direction = path_end_point.theta();
+    if (paritioned_trajectories_ptr->at(i).second ==
+        canbus::Chassis::GEAR_REVERSE) {
+      traj_point_moving_direction =
+          NormalizeAngle(traj_point_moving_direction + M_PI);
+    }
+    double vehicle_moving_direction = vehicle_state.heading();
+    if (vehicle_state.gear() == canbus::Chassis::GEAR_REVERSE) {
+      vehicle_moving_direction =
+          NormalizeAngle(vehicle_moving_direction + M_PI);
+    }
+    // If close to the end point, start on the next trajectory
+    if (distance_to_trajs_end <=
+            config_.open_space_trajectory_partition_config()
+                .kepsilon_to_midway_point() &&
+        std::abs(traj_point_moving_direction - vehicle_moving_direction) <
+            config_.open_space_trajectory_partition_config()
+                .heading_searching_range()) {
+      if (i + 1 >= trajectories_size) {
+        current_trajectory_index = trajectories_size - 1;
+        closest_trajectory_point_index = trajectory_size - 1;
+      } else {
+        current_trajectory_index = i + 1;
+        closest_trajectory_point_index = 0;
+      }
+      flag_change_to_next = true;
+      break;
+    }
+
+    // Choose the closest point to track
+    for (size_t j = 0; j < trajectory_size; ++j) {
+      const TrajectoryPoint& trajectory_point = trajectory.at(j);
+      const PathPoint& path_point = trajectory_point.path_point();
+      double distance = std::sqrt((path_point.x() - vehicle_state.x()) *
+                                      (path_point.x() - vehicle_state.x()) +
+                                  (path_point.y() - vehicle_state.y()) *
+                                      (path_point.y() - vehicle_state.y()));
+      if (distance < min_distance) {
+        min_distance = distance;
+        closest_trajectory_point_index = j;
+      }
+    }
+
+    closest_points.push(std::make_pair(
+        std::make_pair(i, closest_trajectory_point_index), min_distance));
+  }
+
+  if (!flag_change_to_next) {
+    while (!closest_points.empty()) {
+      auto closest_point = closest_points.top();
+      closest_points.pop();
+      double traj_point_moving_direction =
+          paritioned_trajectories_ptr->at(closest_point.first.first)
+              .first.at(closest_point.first.second)
+              .path_point()
+              .theta();
+      if (paritioned_trajectories_ptr->at(closest_point.first.first).second ==
```

Comment:
Give a name to paritioned_trajectories_ptr->at(closest_point.first.first) and reuse it. It's more readable and also shorter.